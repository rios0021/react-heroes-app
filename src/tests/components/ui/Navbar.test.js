import { mount } from "enzyme";
import '@testing-library/jest-dom';
import { MemoryRouter, Router } from "react-router";
import { AuthContext } from "../../../auth/AuthContext";
import { Navbar } from "../../../components/ui/Navbar";
import { types } from "../../../types/types";


describe('Tests in Navbar', () => {

    const historyMock= {
        push: jest.fn(),
        location: {},
        listen: jest.fn(),
        createHref: jest.fn(),
        replace: jest.fn()
    }

    const contextValue = {
        dispatch: jest.fn(),
        user:{
            logged:true,
            name: 'Pedro'
        }
    }
    const wrapper = mount(
        <AuthContext.Provider value={contextValue}>
            <MemoryRouter>
                <Router history={historyMock}>
                    <Navbar/>
                </Router>
            </MemoryRouter>
        </AuthContext.Provider>
    );

    afterEach(()=>{
        jest.clearAllMocks();
    });
    
    test('should show correctly', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.text-info').text()).toBe('Pedro');
    });

    test('should call logout and use history', () => {
        wrapper.find('button').prop('onClick')();
        expect(contextValue.dispatch).toHaveBeenCalledWith({type: types.logout});
        expect(historyMock.replace).toHaveBeenCalledWith('/login');
    });
    
    
})
