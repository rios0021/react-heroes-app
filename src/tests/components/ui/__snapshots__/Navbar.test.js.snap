// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in Navbar should show correctly 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark container-fluid"
>
  <a
    className="navbar-brand"
    onClick={[Function]}
  >
    Heroes
  </a>
  <div
    className="navbar-collapse"
  >
    <div
      className="navbar-nav"
    >
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        Marvel
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        DC
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        Search
      </a>
    </div>
  </div>
  <div
    className="navbar-collapse collapse w-100 order-3 dual-collapse2"
  >
    <ul
      className="navbar-nav ms-auto "
    >
      <span
        className="nav-item nav-link text-info"
      >
        Pedro
      </span>
      <button
        className="btn nav-item nav-link "
        onClick={[Function]}
      >
        Logout
      </button>
    </ul>
  </div>
</nav>
`;
